### PR TITLE
Implemement lft (Linear Fractional Transform) for state-space systems

### PR DESCRIPTION
**Pull request by hungpham2511**
The lft interconnection is implemented. Some test cases, which are generated using matlab's lft functionality, are also included.

### CODE REVIEW

**Comment by hungpham2511** [comment]
Error on testLQR.py. Strange, since the PR does not even touch this part of the code base.

The error, which is reported below, seems similar the error observed in PR https://github.com/python-control/python-control/pull/242  

```
======================================================================
ERROR: testLQR (control.tests.matlab_test.TestMatlab)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/python-control/python-control/control/tests/matlab_test.py", line 418, in testLQR
    np.eye(1), [[1], [1], [2]])
  File "/home/travis/build/python-control/python-control/control/statefbk.py", line 325, in lqr
    X,rcond,w,S,U,A_inv = sb02md(nstates, A_b, G, Q_b, 'C')
  File "/home/travis/miniconda/envs/test-environment/lib/python3.5/site-packages/slycot/synthesis.py", line 350, in sb02md
    raise e
ValueError: the Hamiltonian or symplectic matrix H has less than n stable eigenvalues
```

File: control/statesp.py [hunk 1]
```diff
@@ -611,6 +611,102 @@ def feedback(self, other=1, sign=-1):
...
+        # TODO
+
+        # Figure out the sampling time to use
+        if (self.dt == None and other.dt != None):
+            dt = other.dt       # use dt from second argument
+        elif (other.dt == None and self.dt != None) or \
+                timebaseEqual(self, other):
+            dt = self.dt        # use dt from first argument
+        else:
+            raise ValueError("Systems have different sampling times")
```
Comment (murrayrm) [inline control/statesp.py]:
```suggestion
            raise ValueError("Systems have different time bases")
```

File: control/statesp.py [hunk 2]
```diff
@@ -611,6 +611,102 @@ def feedback(self, other=1, sign=-1):
 
         return StateSpace(A, B, C, D, dt)
 
+    def lft(self, other, nu=-1, ny=-1):
+        """Return the Linear Fractional Transformation.
+
+        See definition here:
+        https://www.mathworks.com/help/control/ref/lft.html
```
Comment (murrayrm) [inline control/statesp.py]: Can we reference a textbook or some other more "definitive" source for the definition of LFT?

**Comment by murrayrm** [comment]
@hungpham2511 Can you rebase this against the current master and also address comments above?

**Comment by hungpham2511** [comment]
Absolutely. I will do that as soon as I have some time.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Pull request merged**